File: testData/src/java8/pkg/TestTrySynchronized.java
package pkg;

import java.io.FileInputStream;
import java.io.InputStream;

public class TestTrySynchronized {
  private static TestTrySynchronized monitor = new TestTrySynchronized();

  public static void case01() throws Exception {

    synchronized(monitor) {
      System.out.println("Inside synchronized block.");
    }

    InputStream stream = null;
    try {
      stream = new FileInputStream("nul");
    } catch(Throwable e) {
      stream.close();
    }
  }

  public static void case02() throws Exception {

    synchronized(monitor) {
      System.out.println("Inside first synchronized block.");
    }

    InputStream stream = null;
    try {
      stream = getInputStream();

      synchronized(monitor) {
        System.out.println("Inside second synchronized block.");
      }

    } catch(Throwable e) {
      stream.close();
    }

  }

  private static InputStream getInputStream() {
    return null;
  }

}

File: plugins/kotlin/testData/src/kt/pkg/TestTrySynchronized.kt
package pkg

import java.io.FileInputStream
import java.io.InputStream

class TestTrySynchronized {

    @Throws(Exception::class)
    fun case01() {
        synchronized(monitor) {
            println("Inside synchronized block.")
        }

        var stream: InputStream? = null
        try {
            stream = FileInputStream("nul")
        } catch (e: Throwable) {
            stream!!.close()
        }
    }

    @Throws(Exception::class)
    fun case02() {
        synchronized(monitor) {
            println("Inside first synchronized block.")
        }

        var stream: InputStream? = null
        try {
            stream = inputStream

            synchronized(monitor) {
                println("Inside second synchronized block.")
            }
        } catch (e: Throwable) {
            stream!!.close()
        }
    }

    private val inputStream: InputStream?
        get() = null

  companion object{
    private val monitor = TestTrySynchronized()
  }
}
